src/bloqade/squin/cirq/lowering.py

```diff
@@ -150,10 +150,79 @@ def lower_measurement(
     ):
         if len(node.qubits) == 1:
             qbit = self.lower_qubit_getindex(state, node.qubits[0])
-            return state.current_frame.push(qubit.MeasureQubit(qbit))
+            stmt = state.current_frame.push(qubit.MeasureQubit(qbit))
+        else:
+            qbits = self.lower_qubit_getindices(state, node.qubits)
+            stmt = state.current_frame.push(qubit.MeasureQubitList(qbits))
 
-        qbits = self.lower_qubit_getindices(state, node.qubits)
-        return state.current_frame.push(qubit.MeasureQubitList(qbits))
+        key = node.gate.key
+        if isinstance(key, cirq.MeasurementKey):
+            key = key.name
+
+        state.current_frame.defs[key] = stmt.result
+        return stmt
+
+    def visit_ClassicallyControlledOperation(
+        self, state: lowering.State[CirqNode], node: cirq.ClassicallyControlledOperation
+    ):
+        conditions: list[ir.SSAValue] = []
+        for outcome in node.classical_controls:
+            key = outcome.key
+            if isinstance(key, cirq.MeasurementKey):
+                key = key.name
+            measurement_outcome = state.current_frame.defs[key]
+
+            if measurement_outcome.type.is_subseteq(ilist.IListType):
+                # NOTE: there is currently no convenient ilist.any method, so we need to use foldl
+                # with a simple function that just does an or
+
+                def bool_op_or(x: bool, y: bool) -> bool:
+                    return x or y
+
+                f_code = state.current_frame.push(
+                    lowering.Python(self.dialects).python_function(bool_op_or)
+                )
+                fn = ir.Method(
+                    mod=None,
+                    py_func=bool_op_or,
+                    sym_name="bool_op_or",
+                    arg_names=[],
+                    dialects=self.dialects,
+                    code=f_code,
+                )
+                f_const = state.current_frame.push(py.constant.Constant(fn))
+                init_val = state.current_frame.push(py.Constant(False)).result
+                condition = state.current_frame.push(
+                    ilist.Foldl(f_const.result, measurement_outcome, init=init_val)
+                ).result
```

Comment:
@Roger-luo Please take a look at this part here specifically. I'm not sure if I'm adding the function here correctly.